src/base/name/traits.rs

```diff
@@ -343,6 +343,35 @@ pub trait ToName: ToLabelIter {
             labels.count() as u8
         }
     }
+
+    fn fmt_with_dot(&self) -> DisplayWithDot<'_, Self> {
+        DisplayWithDot(self)
+    }
+}
+
+pub struct DisplayWithDot<'a, T: ?Sized>(&'a T);
+
+impl<'a, T> fmt::Display for DisplayWithDot<'a, T>
+where
+    T: ToLabelIter + ?Sized,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut labels = self.0.iter_labels();
+        let first = match labels.next() {
+            Some(first) => first,
+            None => unreachable!("at least 1 label must be present"),
+        };
```

Comment:
Use `.expect()`?

Reply:
You'd be right if this was a `panic!`, but `unreachable!` has a slightly different meaning.

Reply:
Hm, that's fair, I suppose.  I wish we had a more concise syntax for this.